src/drivers/MySQLDriver.ts

```diff
@@ -19,27 +18,30 @@ export class MySQLDriver implements IDriver {
     }
 
     private checkConnection(): void {
-        if (this.conn == null)
+        if (!this.conn) {
             throw new Error("MySQL not connected to the database");
+        }
     }
 
     async connect(): Promise<void> {
-        this.conn = (await this.mysql.createPool(this.config)) as any;
+        this.conn = await this.mysql.createPool(this.config);
     }
 
     async prepare(table: string): Promise<void> {
         this.checkConnection();
 
-        await this.conn?.query(
+        await this.conn.query(
```

Comment:
For those, even thought checkConnection is used typescript still see it as possibly underfined. Keeping the optional chaining operator here is needed (this is applicable to all other use of this.conn, I won't make other comments for them thought)
```suggestion
        await this.conn?.query(
```